File: prime_number_primer/prime_number_primer__is_prime_multi.cs
// 実行時間 250ms
// 毎回計算すると遅いのでエラトステネスの篩を使う
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        const int x = 6000000;
        int n = int.Parse( ReadLine() );
        bool[] p = Repeat( true, x + 1 ).ToArray();

        SetOut( new System.IO.StreamWriter( OpenStandardOutput() ) { AutoFlush = false } );
        p[0] = false;
        p[1] = false;
        for( int i = 2; i <= x; i++ ) {
            if( p[i] == true ) {
                for( int j = 2 * i; j <= x; j += i ) {
                    p[j] = false;
                }
            }
        }
        for( int _ = 0; _ < n; _++ ) {
            WriteLine( p[int.Parse( ReadLine() )] ? "pass" : "failure" );
        }
        Out.Flush();
        return;
    }
}

File: prime_number_primer/prime_number_primer__is_prime_multi.py
# 実行時間 2910ms
# 素数判定を毎回行うと遅いのでエラトステネスの篩を使う
import sys

r=sys.stdin.readline
X = 6000000
n = int(r())
p = [True] * (X + 1)

p[0] = False
p[1] = False
for i in range(2, X + 1):
    if p[i] == True:
        j = 2 * i
        while j <= X:
            p[j] = False
            j += i
for _ in [0] * n:
    print("pass" if p[int(r())] else "failure")
